assignment2.c

```diff
@@ -0,0 +1,172 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#define FILENAME "crud.txt"
+
+typedef struct {
+    int id;
+    char name[200];
+    int age;
+} User;
+
+void addUser();
+void listUsers();
+void modifyUser();
+void removeUser();
+void initializeFile();
+
+int main() {
+    int ch;
+
+    initializeFile();
+
+    while (1) {
+        printf("\n--- User Management System ---\n");
+        printf("  1. Add New User\n  2. List All Users\n  3. Modify User by ID\n  4. Remove User by ID\n  5. Exit\n");
+        printf("Enter your choice: ");
+        scanf("%d",&ch);
+        
+        switch (ch) {
+            case 1:
+                addUser();
+                break;
+            case 2:
+                listUsers();
+                break;
+            case 3:
+                modifyUser();
+                break;
+            case 4:
+                removeUser();
+                break;
+            case 5:
+                exit(0);
+            default:
+                printf("Invalid choice. Please try again.\n");
+        }
+    }
+
+    return 0;
+}
+
+void addUser() {
+    User user;
+    FILE *file = fopen(FILENAME, "a");
+
+    if (file == NULL) {
+        printf("File not Found.\n");
+        return;
+    }
+    printf("Enter User ID: ");
+    scanf("%d", &user.id);
+    printf("Enter Name: ");
+    scanf(" %[^\n]s", user.name);
+    printf("Enter Age: ");
+    scanf("%d", &user.age);
+    fprintf(file, "%d %s %d\n", user.id, user.name, user.age);

```

Comment:
Ensure user IDs are unique. When adding or modifying users, you should check if the ID already exists in the file.